src/txmempool.h

```diff
@@ -573,6 +574,8 @@ class CTxMemPool
 
     const Limits m_limits;
 
+    const fs::path m_persistence_path{};
```

Comment:
In commit "mempool: Introduce m_persistence_path member" (ea750b1374fa0a49212b1f280660be7b2f6b9b2f)

I suspect it would simplify the PR and the code overall not to add a `CTxMemPool::m_persistence_path` member or an `MemPoolOptions::persistence_path` option. You could avoid the noise and statefulness and the twisty `const fs::path dump_path{mempool.m_persistence_path.empty() ? MemPoolPersistencePath(args) : mempool.m_persistence_path};` logic by writing a simple helper functions:

```c++
bool MempoolPersist(const ArgsManager& args)
{
    return args.GetBoolArg("-persistmempool", DEFAULT_PERSIST_MEMPOOL));
}

fs::path MempoolPath(const ArgsManager& args)
{
    return args.GetDataDirNet() / "mempool.dat";
}
```

And calling them on as needed from validation and RPC code without adding extra options and state.

Reply:
Mmm, this would simplify the code in question, though I personally prefer the pattern of parsing these upfront to avoid having to involve ArgsManager for something that seems like a property of `CTxMemPool`.

Reply:
Implemented as of 80b303ea1a065602cf3f84d45b04dce95bfb6952